// qzready/Gruntfile.js
/*
 * qzready
 * https://bitbucket.org/quartz/qzready
 *
 * Copyright (c) 2014 Richard Stovall
 * Not licensed for public use
 */

'use strict';

module.exports = function(grunt) {

	require('load-grunt-tasks')(grunt);

	grunt.initConfig({
		pkg: grunt.file.readJSON('package.json'),

		clean: ['tmp'],
		copy: {
			styles: {
				src: ['css/style.css'],
				dest: 'tmp/style.css'
			}
		},
		replace: {
			styles: {
				src: ['tmp/style.css'],
				dest: 'tmp/style.css'
			},
			index: {
				src: ['tmp/index.html'],
				dest: 'tmp/index.html'
			},
			source: {
				src: ['tmp/index.html'],
				dest: 'tmp/index.html'
			}
		},
		cssmin: {
			styles: {
				src: ['tmp/style.css'],
				dest: 'tmp/style.css'
			}
		},
		concat: {
			scripts: {
				options: {
					seperator: '\n'
				},
				dest: 'tmp/app.js'
			},
			styletags: {
				options: {
					banner: '<style type="text/css">',
					footer: '</style>'
				},
				src: 'tmp/style.css',
				dest: 'tmp/style.css'
			},
			scripttags: {
				options: {
					banner: '<script type="text/javascript">',
					footer: '</script>'
				},
				src: 'tmp/app.js',
				dest: 'tmp/app.js'
			},
			addscripts: {
				src: 'tmp/index.html',
				dest: 'tmp/index.html'
			},
			create: {
				src: ['tmp/style.css','tmp/index.html','tmp/app.js']
			}
		},
		uglify: {
			options: {
				mangle: false
			},
			my_target: {
				files: {
					'tmp/app.js': ['tmp/app.js']
				}
			}
		},
		prettify: {
			options: {
				indent: 1,
				indent_char: "	",
				preserve_newlines: true,
				unformatted: ['a', 'sub', 'sup', 'b', 'i', 'u']
			},
			index: {
				src: 'tmp/index.html',
				dest: 'tmp/index.html'
			}
		},
		qzready: {
			deploy: {
				options: {
					base: 'http://ads.qz.com/sponsors',
					client: 'vmware',
					campaign: 'sddc-2',
					date: '2014-04-10',
					unit: 'desktop',
					version: '1',
					internal_scripts: [
						
					],
					external_scripts: [
						'http://app.qz.com/js/vendor/jQuery-min.js',
						'http://admin.brightcove.com/js/BrightcoveExperiences.js',
						'http://app.qz.com/js/frameMessager/min/frameMessager.min.3.0.2.js',
						'http://app.qz.com/js/frameMessager/QZIX.js'
					]
				}
			}
		}
	});

	grunt.loadTasks('tasks');

	grunt.registerTask('default', ['qzready']);
};
